src/Datadog.Trace.ClrProfiler.Managed/Emit/ObjectExtensions.cs

```diff
@@ -271,5 +271,40 @@ private static string GetKey<TResult>(string name, Type type)
             dynamicMethod.Return();
             return dynamicMethod.CreateDelegate();
         }
+
+        private readonly struct PropertyFetcherCacheKey : IEquatable<PropertyFetcherCacheKey>
```

Comment:
I'm using the same key type for methods and properties. As far as I know, a type can't have a property and a method with the same name, so I don't believe collisions are possible